src/vue/forms/AtomicSwapQuoteAssetsForm.vue

```diff
@@ -0,0 +1,292 @@
+              :disabled="isDisabled"
+            />
+            <!-- eslint-enable max-len -->
+          </div>
+        </div>
+      </div>
+      <div
+        class="atomic-swap-quote-assets-form__add-quote-asset-wrp"
+        v-if="canAddQuoteAsset(index + 1)"
+      >
+        <p class="atomic-swap-quote-assets-form__add-quote-asset">
+          <button
+            class="atomic-swap-quote-assets-form__add-quote-asset-btn"
+            type="button"
+            @click="addQuoteAsset(index)"
+            :disabled="isDisabled"
+          >
+            <!-- eslint-disable-next-line max-len -->
+            {{ 'atomic-swap-quote-assets-form.add-quote-asset-btn' | globalize }}
+          </button>
+        </p>
+      </div>
+    </div>
+
+    <div class="app__form-actions">
+      <button
+        v-ripple
+        class="atomic-swap-quote-assets-form__submit-btn app__button-raised"
+        :disabled="isDisabled"
+      >
+        {{ 'atomic-swap-quote-assets-form.create-btn' | globalize }}
+      </button>
+    </div>
+  </form>
+</template>
+
+<script>
+import FormMixin from '@/vue/mixins/form.mixin'
+import {
+  required,
+  selectedSameAssetCode,
+  cardNumber,
+  address,
+} from '@validators'
+import { mapGetters } from 'vuex'
+import { vuexTypes } from '@/vuex'
+import { globalize } from '@/vue/filters/globalize'
+
+const EVENTS = {
+  submit: 'submit',
+}
+
+export default {
+  name: 'atomic-swap-quote-assets-form',
+  mixins: [FormMixin],
+  props: {
+    isDisabled: { type: Boolean, default: false },
+  },
+  data: _ => ({
+    form: {
+      quoteAssets: [{ price: '', asset: {}, destination: '' }],
+    },
+    isLoaded: false,
+    isLoadFailed: false,
+    isFormSubmitting: false,
+  }),
+
+  validations () {
+    return {
+      form: {
+        quoteAssets: {
+          $each: {
+            asset: {
+              required,
+              selectedSameAssetCode: asset => {
+                /* eslint-disable max-len */
+                const selectedAssets = this.getSelectedAssetsByAssetCode(asset.code)
+                return selectedSameAssetCode(selectedAssets)
+                /* eslint-enable max-len */
+              },
+            },
+            destination: {
+              required,
+              addressOrNumberCard: (value, quoteAsset) => {
+                return quoteAsset.asset.isCoinpayments
+                  ? address(quoteAsset.asset.code)(value)
+                  : cardNumber(value)
+              },
+            },
+          },
+        },
+      },
+    }
+  },
+
+  computed: {
+    ...mapGetters({
+      quoteAtomicSwapAssets: vuexTypes.quoteAtomicSwapAssets,
+    }),
+  },
+
+  async created () {
+    this.form.quoteAssets[0].asset = this.quoteAtomicSwapAssets[0] || {}
+  },
+  methods: {
+    submit () {
+      this.$emit(EVENTS.submit, this.form)
+    },
+
+    getSelectedAssetsByAssetCode (assetCode) {
+      const selectedAssetsCode = []
+      this.form.quoteAssets.forEach(quoteAsset => {
```

Comment:
please use `reduce` and return number of repeat

Reply:
used